scripts/globals/magic_utils/spell_damage.lua

```diff
@@ -0,0 +1,790 @@
+
+        -----------------------------------
+        -- Apply level correction.
+        -----------------------------------
+        local levelDiff =  utils.clamp(caster:getMainLvl() - target:getMainLvl(), -5, 5)
+        magicAcc = magicAcc + levelDiff * 3
+
+        -----------------------------------
+        -- STEP 2: Get target magic evasion
+        -- Base magic evasion (base magic evasion plus resistances(players), plus elemental defense(mobs)
+        -----------------------------------
+        magicEva = target:getMod(xi.mod.MEVA) + resMod
+
+        -----------------------------------
+        -- STEP 3: Get Magic Hit Rate
+        -- https://www.bg-wiki.com/ffxi/Magic_Hit_Rate
+        -----------------------------------
+        local magicAccDiff = magicAcc - magicEva
+
+        if magicAccDiff < 0 then
+            magicHitRate = utils.clamp(50 + math.floor(magicAccDiff / 2), 5, 95)
+        else
+            magicHitRate = utils.clamp(50 + magicAccDiff, 5, 95)
+        end
+    end
+
+    -----------------------------------
+    -- STEP 4: Get Resist Tier
+    -----------------------------------
+    if skillchainCount == 0 then -- Magic bursts do not get resisted.
+        local resistTier = 0
+        local randomVar  = 0
+
+        for i = 3, 1, -1 do
+            randomVar = math.random(1, 100)
+            if randomVar > magicHitRate then
+                resistTier = resistTier + 1
+            end
+        end
+
+        if resistTier == 0 then     -- Unresisted
+            resist = 1.0
+        elseif resistTier == 1 then -- (1/2)
+            resist = 0.5
+        elseif resistTier == 2 then -- (1/4)
+            resist = 0.25
+        elseif resistTier == 3 then -- (1/8)
+            resist = 0.125
+        end
+    else
+        resist = 1.0
+    end
+
+    return resist
+end
+
+xi.magic_utils.spell_damage.calculateIfMagicBurst = function(caster, target, spell, spellElement)
+    local magicBurst                      = 1 -- The variable we want to calculate
+    local skillchainTier, skillchainCount = FormMagicBurst(spellElement, target) -- External function. Not present in magic.lua.
+
+    if skillchainCount > 0 then
+        magicBurst = 1.25 + (0.1 * skillchainCount) -- Here we add SDT DAMAGE bonus for magic bursts aswell, once SDT is implemented. https://www.bg-wiki.com/ffxi/Resist#SDT_and_Magic_Bursting
+    end
+
+    return magicBurst
+end
+
+xi.magic_utils.spell_damage.calculateIfMagicBurstBonus = function(caster, target, spell, spellId, spellElement)
+    local magicBurstBonus                 = 1.0 -- The variable we want to calculate
+    local modBurst                        = 1.0
+    local ancientMagicBurstBonus          = 0
+    local skillchainTier, skillchainCount = FormMagicBurst(spellElement, target) -- External function. Not present in magic.lua.
+
+    if spellId >= 204 and spellId <= 215 then -- If spell is Ancient Magic
```

Comment:
I think we have spell names enum'd somewhere now so that we don't have to have the ID use